Gemfile

```diff
@@ -13,6 +13,16 @@ gem "ohai", git: "https://github.com/chef/ohai.git", branch: "master"
 gem "chef-config", path: File.expand_path("../chef-config", __FILE__) if File.exist?(File.expand_path("../chef-config", __FILE__))
 gem "cheffish", "~> 14"
 
+# The chef_core gems are sourced from git.
+# gem "chef_core", git: "https://github.com/chef/chef_core", branch: "chef-core-split"
+# gem "chef_core-actions", git: "https://github.com/chef/chef_core-actions", branch: "chef-core-split"
+# gem "chef_core-cliux", git: "https://github.com/chef/chef_core-cliux", branch: "chef-core-split"
+# Temporary for testing:
+gem "train", path: "../train"
+gem "chef_core", path: "../chef_core"
+gem "chef_core-actions", path: "../chef_core-actions"
+gem "chef_core-cliux", path: "../chef_core-cliux"
```

Comment:
Few issues some trivial some not so much:

- should not use underscores in gem names.
- should probably not be immediately extracting out 3 different dependent gems if possible, i'd prefer to see 1 or may be 2
- the "chef core" name is probably poor since it suggests to me something that its not clear to me that that is
- the dependent gems MUST go into the chef repo itself similar to the chef-config gem, i can push some code i was working on for the chef-helpers gem which can show how.  the way forwards for sub-gems of chef client needs to be more monorepoish and lockstepping with the versions, the mixlib style is poor (talk to @tas50 about the joys of major version bumping the universe which should be freshly burned into his brain).   this is very, very hard 'no' on the direction here with additional gem repos.
- since knife is requiring chef_core files it needs to appear in the gemspec.  that looks like it ultimately creates circular deps here:  https://github.com/chef/chef_core-actions/blob/master/chef_core-actions.gemspec#L40-L42 which is either a mistake or indicates  problem in modelling how this is being extracted.

its not "no don't do this" but i have deep concerns about the way that this is being done.

Reply:
Some READMEs in the dependent gems or a few paragraphs of a design doc somewhere explaining what and why would also help.

Reply:
The circular dep is a mistake - will get that straightened out.  I am +1 on merging these into a single repo with multiple gemspecs.

Reply:
I just saw that I didn't address these in this discussion.  To that end: 

* gem name changed to `chef-core`, with `chef-core-actions` as optional. chef-core-cliux is now just part of chef-core and doesn't have a separate gem. The repos for -cliux and -actions have been removed. 
* it looks like that chef dep was in chef-core-actions, which was in the Gemfile - but not the gemspec.  It shouldn't have been there -- we don't use it in bootstrap and it does have the dep chef-core-actions -> chefdk -> chef. I've confirmed that knife + chef-core does not do this. 
* We'll be  updating  the chef-core repo with a README in the coming days.

Reply:
Looking a little closer, even chef-core-actions should not have that dep - it lives only in chef-apply. chef-core-actions and chef-apply have been updated to reflect this.  `chef-core` now has no lower-level chef deps except `chef-config`. (It only uses that to get a one-line function, so that can go too if it becomes a problem.)

Reply:
yeah so chef can can depend on chef-core and chef-config with chef-core also depending on chef-config since that directed graph does not have a circular dep.

if you potentially start looking at chef-config needing a dep on chef-core then you'd need to remove the dep going the other direction and do something else with that function.